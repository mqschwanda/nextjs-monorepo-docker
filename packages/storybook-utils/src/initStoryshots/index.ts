import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
