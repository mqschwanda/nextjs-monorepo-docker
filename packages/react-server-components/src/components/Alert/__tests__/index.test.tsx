import { render } from '@testing-library/react';
import path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import { Alert } from '@';

initStoryshots({
  configPath: path.resolve(__dirname, '../../../../../storybook/.storybook'),
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Alert />', () => {
      it('renders', () => {
        const testId = Alert.name;
        const chilren = 'children...';

        const { getByTestId, asFragment } = render((
          <Alert
            className='test'
            testId={testId}
            variantBackgroundColor='info'
          >
            { chilren }
          </Alert>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
