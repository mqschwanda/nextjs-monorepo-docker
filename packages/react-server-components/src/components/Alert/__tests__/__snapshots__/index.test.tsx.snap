// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Alert /> renders 1`] = `
<DocumentFragment>
  <div
    class="alert alert-info test"
    data-testid="Alert"
  >
    children...
  </div>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Alert Alert Error Example 1`] = `
<div
  className="alert alert-error"
  data-testid="Alert"
>
  <div>
    <svg
      className="stroke-current flex-shrink-0 h-6 w-6"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <span>
      Error! Task failed successfully.
    </span>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Alert Alert Example 1`] = `
<div
  className="alert"
  data-testid="Alert"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut mollis.
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Alert Alert Info Example 1`] = `
<div
  className="alert alert-info"
  data-testid="Alert"
>
  <div>
    <svg
      className="stroke-current flex-shrink-0 w-6 h-6"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <span>
      New software update available.
    </span>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Alert Alert Success Example 1`] = `
<div
  className="alert alert-success"
  data-testid="Alert"
>
  <div>
    <svg
      className="stroke-current flex-shrink-0 h-6 w-6"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <span>
      Your purchase has been confirmed!
    </span>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Alert Alert Warning Example 1`] = `
<div
  className="alert alert-warning"
  data-testid="Alert"
>
  <div>
    <svg
      className="stroke-current flex-shrink-0 h-6 w-6"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <span>
      Warning: Invalid email address!
    </span>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Alert Alert With Buttons Example 1`] = `
<div
  className="alert"
  data-testid="Alert"
>
  <svg
    className="stroke-info flex-shrink-0 w-6 h-6"
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </svg>
  <span>
    we use cookies for no reason.
  </span>
  <div>
    <button
      className="btn btn-ghost btn-sm"
      data-testid="Button"
      disabled={false}
      type="button"
    >
      Deny
    </button>
    <button
      className="btn btn-primary btn-sm"
      data-testid="Button"
      disabled={false}
      type="button"
    >
      Accept
    </button>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Alert Alert With Title And Description Example 1`] = `
<div
  className="alert"
  data-testid="Alert"
>
  <svg
    className="stroke-info flex-shrink-0 w-6 h-6"
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </svg>
  <div>
    <h3
      className="font-bold"
    >
      New message!
    </h3>
    <div
      className="text-xs"
    >
      You have 1 unread message
    </div>
  </div>
  <div
    className="flex-none"
  >
    <button
      className="btn btn-sm"
      data-testid="Button"
      disabled={false}
      type="button"
    >
      See
    </button>
  </div>
</div>
`;
